src/tts_text/phoneme_covering_set.py

```diff
@@ -0,0 +1,196 @@
+    # paragraphs, etc.
+    document_text = document["text"]
+    " ".join(
+        re.split(
+            "\t|\n|_NEWLINE_|_START_ARTICLE_|_START_SECTION_|_START_PARAGRAPH_",
+            document["text"],
+        )
+    )
+    document_text = re.sub(" +", " ", document_text).strip()
+    words = document_text.split(" ")
+
+    counter = Counter(words)
+    for language, phoneme_list in phonemes.items():
+        phoneme_count = 0
+        found_phonemes = []
+        found_example_words = []
+        for phoneme in phoneme_list:
+            for word in phoneme["examples"]:
+                occurences = counter[word]
+                phoneme_count += occurences
+                if occurences > 0:
+                    found_phonemes.append(phoneme["name"])
+                    found_example_words.append(word)
+        document[f"{language}_phoneme_count"] = phoneme_count
+        document[f"{language}_phonemes"] = found_phonemes
+        document[f"{language}_found_example_words"] = found_example_words
+        document[f"{language}_unique_phonemes"] = set(found_phonemes)
+        document[f"{language}_unique_phonemes_count"] = len(set(found_phonemes))
+
+    document["all_phoneme_count"] = (
+        document["da_phoneme_count"] + document["en_phoneme_count"]
+    )
+    document["all_phonemes"] = document["da_phonemes"] + document["en_phonemes"]
+    document["all_found_example_words"] = (
+        document["da_found_example_words"] + document["en_found_example_words"]
+    )
+    document["all_unique_phonemes"] = set(
+        document["da_unique_phonemes"] | document["en_unique_phonemes"]
+    )
+    document["all_unique_phonemes_count"] = len(document["all_unique_phonemes"])
+    return document
+
+
+def make_phoneme_covering_set(cfg: DictConfig, output_path: Path) -> list[str]:
+    """Create a phoneme covering set from the sorted wiki dataset.
+
+    A phoneme covering set is a set of strings that contains at least one example
+    word for each phoneme.
+
+    Args:
+        cfg (DictConfig): The Hydra configuration object.
+
+    Returns:
+        list[str]: The phoneme covering set.
+    """
+    sorted_dataset = sort_wiki_by_phoneme_occurence(cfg=cfg)
+
+    raw_data_path = Path(cfg.dirs.data) / "raw"
+    phoneme_path = raw_data_path / cfg.phoneme_filename
+    with open(phoneme_path) as f:
+        phonemes = json.load(f)
+    all_phone_names = [entry["name"] for entry in phonemes["da"]] + [
+        entry["name"] for entry in phonemes["en"]
+    ]
+
+    covering_set = []
+    for document in tqdm(sorted_dataset):
+        document_phonemes = document["all_phonemes"]
+        new_phoneme_found = False
+        for phoneme in document_phonemes:
+            if phoneme in all_phone_names:
+                new_phoneme_found = True
+                all_phone_names.pop(all_phone_names.index(phoneme))
+        if new_phoneme_found:
+            covering_set.append(document)
+        if len(all_phone_names) == 0:
+            break
```

Comment:
Slight simplification, but mostly a matter of taste, so just ignore it if you don't like it.

```suggestion
    all_phone_names = {entry["name"] for entry in phonemes["da"]}.union(
        {entry["name"] for entry in phonemes["en"]}
    )

    covering_set = []
    for document in tqdm(sorted_dataset):
        if not all_phone_names:
            break
        document_phonemes = document["all_phonemes"]
        new_phonemes = set(document_phonemes).difference(all_phone_names)
        for new_phoneme in new_phonemes:
            all_phone_names.pop(phoneme)
        if new_phonemes:
            covering_set.append(document)
```